osu.Game/Overlays/Chat/ChannelList/ChannelList.cs

```diff
@@ -0,0 +1,161 @@
+                            },
+                            privateChannelFlow = new ChannelListItemFlow("DIRECT MESSAGES"),
+                        },
+                    },
+                },
+            };
+        }
+
+        public void AddChannel(Channel channel)
+        {
+            if (channelMap.ContainsKey(channel))
+                return;
+
+            ChannelListItem item = new ChannelListItem(channel);
+            item.OnRequestSelect += chan => OnRequestSelect?.Invoke(chan);
+            item.OnRequestLeave += chan => OnRequestLeave?.Invoke(chan);
+
+            ChannelListItemFlow flow = getFlowForChannel(channel);
+            channelMap.Add(channel, item);
+            flow.Add(item);
+        }
+
+        public void RemoveChannel(Channel channel)
+        {
+            if (!channelMap.ContainsKey(channel))
+                return;
+
+            ChannelListItem item = channelMap[channel];
+            ChannelListItemFlow flow = getFlowForChannel(channel);
+
+            channelMap.Remove(channel);
+            flow.Remove(item);
+        }
+
+        public ChannelListItem GetItem(Channel channel)
+        {
+            if (!channelMap.ContainsKey(channel))
+                throw new ArgumentOutOfRangeException();
+
+            return channelMap[channel];
+        }
+
+        private ChannelListItemFlow getFlowForChannel(Channel channel)
+        {
+            switch (channel.Type)
+            {
+                case ChannelType.Public:
+                    return publicChannelFlow;
+
+                case ChannelType.PM:
+                    return privateChannelFlow;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private class ChannelListItemFlow : FillFlowContainer
+        {
+            public ChannelListItemFlow(string label)
+            {
+                Direction = FillDirection.Vertical;
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+
+                Add(new OsuSpriteText
+                {
+                    Text = label,
+                    Margin = new MarginPadding { Left = 18, Bottom = 5 },
+                    Font = OsuFont.Torus.With(size: 12, weight: FontWeight.SemiBold),
+                });
+            }
+        }
+
+        private class ChannelListScrollContainer : OsuScrollContainer
+        {
+            protected override ScrollbarContainer CreateScrollbar(Direction direction)
+                => new ChannelListScrollBar(direction);
+
+            protected class ChannelListScrollBar : OsuScrollbar
+            {
+                private const float bar_size = 4;
+                private const float bar_margin = 7;
+
+                public ChannelListScrollBar(Direction scrollDir)
+                    : base(scrollDir)
+                {
+                    Size = new Vector2(bar_size);
```

Comment:
Please use the original size of scrollbars elsewhere in the game, there's no need to make this so thin I can't select it.

@arflyte are all scrollbars to be this thin in the future? If so, it should be updated in a separate PR in `OsuScrollContainer`, and padding out the clickable area to still be as large as it used to be.

Reply:
Didn't intentionally design a new scrollbar. Use what we currently have.